### PR TITLE
Week3 step5 Use datastore to make comments permanent

### PR DESCRIPTION
**Pull request by Mvalverde00**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -27,23 +36,52 @@
...
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(comments));
   }
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json = gson.toJson(comments);
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String newComment = request.getParameter("comment");
+    if (newComment != null && newComment != "") {
```
Comment (qlyao) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:60]: `!newComment.isEmpty()`

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -27,23 +36,52 @@
...
+      // The idea of parent and score will be fleshed out later.
```
Comment (qlyao) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:61]: nit: use `TODO` for such comments

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+ * @param date_posted The time the comment was submitted
+ * @param parent The id of the comment's parent.  0 if top level comment
+ * @param score A rating of the comment, decided by users' upvotes/downvotes
+
+ */
+public final class Comment {
+
+  private final long id;
+  private final String message;
+  private final long date_posted;
```
Comment (ChuxiangP) [inline portfolio/src/main/java/com/google/sps/data/Comment.java:32]: nit: please use camel case for member variables: https://google.github.io/styleguide/javaguide.html#s5.2.5-non-constant-field-names

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    this.message = message;
+    this.date_posted = date_posted;
+    this.parent = parent;
+    this.score = score;
+  }
+
+  public Comment(Entity e) {
+    this(e.getKey().getId(), (String) e.getProperty("message"),
+        (long) e.getProperty("date_posted"), (long) e.getProperty("parent"),
+        (long) e.getProperty("score"));
```
Comment (ChuxiangP) [inline portfolio/src/main/java/com/google/sps/data/Comment.java:47]: nit: let's keep one argument per line.

```
this(
    e.getKey().getId(),
    (String) e.getProperty("message"),
    (long) e.getProperty("date_posted"),
    (long) e.getProperty("parent"),
    (long) e.getProperty("score"));
```

**Pull request merged**